Formula/cdo.rb

```diff
@@ -0,0 +1,30 @@
+class Cdo < Formula
+  desc "Climate Data Operators"
+  homepage "https://code.mpimet.mpg.de/projects/cdo"
+  url "https://code.mpimet.mpg.de/attachments/download/20826/cdo-1.9.8.tar.gz"
+  sha256 "f2660ac6f8bf3fa071cf2a3a196b3ec75ad007deb3a782455e80f28680c5252a"
+
+  depends_on "eccodes"
+  depends_on "hdf5"
+  depends_on "netcdf"
+  depends_on "proj"
+  depends_on "szip"
+
+  def install
+    args = %W[
+      --disable-dependency-tracking
+      --prefix=#{prefix}
+      --with-eccodes=#{Formula["eccodes"].opt_prefix}
+      --with-netcdf=#{Formula["netcdf"].opt_prefix}
+      --with-hdf5=#{Formula["hdf5"].opt_prefix}
+      --with-szlib=#{Formula["szip"].opt_prefix}
+    ]
+
+    system "./configure", *args
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/cdo", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
As this utility is all about binary numerical formats I would have to uuencode one of the test cases (the individual files are not available for download, only the complete package) and use that from the test case. But this would blow up the formula considerably. Is that intended?

Reply:
You could just download the complete package and pick one I guess? While it's obviously not the intention to make the formulae larger, I think everyone can agree that it's a small price to pay to make sure we don't ship broken software.